xds/internal/balancer/xdsrouting/matcher_path.go

```diff
@@ -30,14 +30,26 @@ type pathMatcherInterface interface {
 }
 
 type pathExactMatcher struct {
-	fullPath string
+	// fullPath is all upper case if caseInsensitive is true.
+	fullPath        string
+	caseInsensitive bool
 }
 
-func newPathExactMatcher(p string) *pathExactMatcher {
-	return &pathExactMatcher{fullPath: p}
+func newPathExactMatcher(p string, caseInsensitive bool) *pathExactMatcher {
+	ret := &pathExactMatcher{
+		fullPath:        p,
+		caseInsensitive: caseInsensitive,
+	}
+	if caseInsensitive {
+		ret.fullPath = strings.ToUpper(p)
+	}
+	return ret
 }
 
 func (pem *pathExactMatcher) match(path string) bool {
+	if pem.caseInsensitive {
+		return pem.fullPath == strings.ToUpper(path)
```

Comment:
How about using [EqualFold](https://golang.org/pkg/strings/#EqualFold) instead of storing the path in all upper case when `caseInsensitive` is used?